src/index.js

```diff
@@ -55,30 +55,28 @@ export default function workerize(code) {
 		};
 		*/
 		ctx.addEventListener('message', ({ data }) => {
-			if (data.type==='RPC') {
-				let id = data.id;
-				if (id!=null) {
-					if (data.method) {
-						let method = rpcMethods[data.method];
-						if (method==null) {
-							ctx.postMessage({ type: 'RPC', id, error: 'NO_SUCH_METHOD' });
-						}
-						else {
-							Promise.resolve()
-								.then( () => method.apply(null, data.params) )
-								.then( result => { ctx.postMessage({ type: 'RPC', id, result }); })
-								.catch( error => { ctx.postMessage({ type: 'RPC', id, error }); });
-						}
-					}
-					else {
-						let callback = callbacks[id];
-						if (callback==null) throw Error(`Unknown callback ${id}`);
-						delete callbacks[id];
-						if (data.error) callback.reject(Error(data.error));
-						else callback.resolve(data.result);
-					}
+			if (data.type!=='RPC') return;
+			let id = data.id;
+			if (id==null) return;
+			if (data.method) {
+				let method = rpcMethods[data.method];
+				if (method==null) {
+					ctx.postMessage({ type: 'RPC', id, error: 'NO_SUCH_METHOD' });
+					return;
 				}
+
+				Promise.resolve(() => method.apply(null, data.params))
```

Comment:
This creates a Promise with a value resolved to the function itself. See below:
<img width="415" alt="screen shot 2018-01-11 at 10 01 02 am" src="https://user-images.githubusercontent.com/105127/34831922-67a48690-f6b6-11e7-9ed8-0c97e8ccb061.png">